xbmc/windowing/osx/WinSystemOSX.mm

```diff
@@ -1180,36 +1251,31 @@ static void DisplayReconfigured(CGDirectDisplayID display,
     oldRefreshRate = m_refreshRate;
 
     // send a message so that videoresolution (and refreshrate) is changed
-    NSWindow* win = m_appWindow;
-    NSRect frame = win.contentView.frame;
-    CServiceBroker::GetAppMessenger()->PostMsg(TMSG_VIDEORESIZE, frame.size.width,
-                                               frame.size.height);
+    dispatch_sync(dispatch_get_main_queue(), ^{
+      NSWindow* win = m_appWindow;
+      NSRect frame = win.contentView.frame;
+      CServiceBroker::GetAppMessenger()->PostMsg(TMSG_VIDEORESIZE, frame.size.width,
+                                                 frame.size.height);
+    });
+  }
+  // store window position in window mode
+  if (!m_bFullScreen)
+  {
+    dispatch_sync(dispatch_get_main_queue(), ^{
+      settings->SetInt(SETTING_WINDOW_LEFT, m_appWindow.frame.origin.x);
+      settings->SetInt(SETTING_WINDOW_TOP, m_appWindow.frame.origin.y);
+      settings->Save();
+    });
```

Comment:
hope it's fine to work with settings from non-XBMC thread

Reply:
usually not but mainly because of settings callbacks when they are changed. In this case they are just dummy settings to cache the values, so it's safe.
Regardless I hope this can be dropped in the future with `NSWindowController` as suggested above. Commits have low granularity in this PR so it's easy to revert